### ISSUE TITLE
Invalid imput

### ISSUE THREAD

**Issue by Coinlogru** [nboy1/Admob-for-Flash-ANE-GMA21-GLM-#1]
Hello!
In the manual not declared how to import path to *.ane file in Adobe Animate ActionScript settings, because without it build error 1120: undefined Admob.
Also if path declared and when app bulid for iOS, it's anyway return error "Invalid imput."
So, even can't test that ANE.

**Comment by nboy1** [comment]
> Hello! In the manual not declared how to import path to *.ane file in Adobe Animate ActionScript settings, because without it build error 1120: undefined Admob. Also if path declared and when app bulid for iOS, it's anyway return error "Invalid imput." So, even can't test that ANE.

hi @Coinlogru 
as i understood you are talking about import ANE in Adobe Animate and not "    import com.glm.admob.*;   "?
if so this how to import it in adobe animate 
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/39934716/e31b9452-ef9b-421a-b78f-e7041e79f823)

**Comment by Coinlogru** [comment]
That is correct, but path in the zip is /Admob for Flash ANE GMA21 (GLM)/Admob_ANE_ios_android/Admob-Android-IOS-v1.ane

Also "Invalid imput." error coming not only with your ane, it is also happen when wrong path to iOS Simulator, etc. So, does a path in the zip are correct?

**Comment by nboy1** [comment]
all matter is ane download this ane  https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/releases/tag/v1.0.0
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/39934716/5b58ab01-93f5-4367-a1cf-951c5daba413)

and put it where ever you want and select it from the selector in adobe animate, it doesn't matter the path you put it in just be sure that you selected it and imported it in animate

**Comment by Coinlogru** [comment]
Here is a path:
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/110308598/8525dbb4-a9fb-476a-ac09-c32c227a2816)

Here is a .as file:
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/110308598/7a1be6cd-a6b3-4053-8444-2e4e2204613a)

Here is a scripts:
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/110308598/60054a7e-c953-461b-84ff-20800834852c)
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/110308598/5f217d56-33aa-4b80-a793-f7c0d8ca7a4e)

And here is error:
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/110308598/9d2c21e1-50ba-493a-a390-a6908925643b)

For Android it compile without errors, but ads not showing, even test ads.

**Comment by Coinlogru** [comment]
How to do this?
https://github.com/lilili87222/admob-for-flash/issues/12#issuecomment-1127445017
https://github.com/lilili87222/admob-for-flash/issues/12#issuecomment-1127447428
I have Mac and Apache ANT is installed.
I have create new library in Android Studio
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/110308598/e9471550-3490-4c91-b786-f8f0e77a84a0)
What mean: "copy java file to src folder"? What java file?
Also where to download Google Mobile Ads SDK files and where to copy it?
Can't find it there: https://developers.google.com/admob/android/quick-start

**Comment by nboy1** [comment]
> Here is a path: ![image](https://user-images.githubusercontent.com/110308598/265852074-8525dbb4-a9fb-476a-ac09-c32c227a2816.png)
> 
> Here is a .as file: ![image](https://user-images.githubusercontent.com/110308598/265851800-7a1be6cd-a6b3-4053-8444-2e4e2204613a.png)
> 
> Here is a scripts: ![image](https://user-images.githubusercontent.com/110308598/265852422-60054a7e-c953-461b-84ff-20800834852c.png) ![image](https://user-images.githubusercontent.com/110308598/265852586-5f217d56-33aa-4b80-a793-f7c0d8ca7a4e.png)
> 
> And here is error: ![image](https://user-images.githubusercontent.com/110308598/265851983-9d2c21e1-50ba-493a-a390-a6908925643b.png)
> 
> For Android it compile without errors, but ads not showing, even test ads.



first comment:
don't show Rewarded Video inside the events this mean you will show ads every time ads loaded, if you are just testing i think it's better to put it in a timer or something.

second comment:
i think you are talking about lilili87222 source code? i didn't publish lilili87222 Android updated code,  i asked if some one could update ios part because i couldn't catch up or follow his steps in ios part, that's why i started the new GLM ANE from scratch. you can use GLM code source.

invalid input : i think it is in ios, i think maybe something with ios certificates try to export game without admob ANE and see if you still have that error. if you have it mean something have no relation with the ANE.

"What mean: "copy java file to src folder"? What java file?
Also where to download Google Mobile Ads SDK files and where to copy it?":

i need tutorial for this to explain it " there is a lot of java files and jar's some from android project and sdk jars from Maven repository and need modifications as described in comments there(a lot of work) ,you can follow comments there while trying to update it i explained a lot of steps there, or download the GLM source code and try to figure out how things made and components of every folder  you will get an idea.

if you want the updated  lilili87222 source code just give me where to send it to you, but i think have no meaning because it is only Android part.

**Comment by Coinlogru** [comment]
Here is outside event, but ads not showing at Android
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/110308598/6b9da56f-b457-4d17-9079-e5279bfd7a7c)

When I compile the same app with lilili87222 ANE for IOS it is install without errors and works ok, and then test ads are showing, but other ads not showing.

Yes, please send me to coinlogru@gmail.com

**Comment by nboy1** [comment]
> Here is outside event, but ads not showing at Android ![image](https://user-images.githubusercontent.com/110308598/265991119-6b9da56f-b457-4d17-9079-e5279bfd7a7c.png)
> 
> When I compile the same app with lilili87222 ANE for IOS it is install without errors and works ok, and then test ads are showing, but other ads not showing.
> 
> Yes, please send me to [coinlogru@gmail.com](mailto:coinlogru@gmail.com)

Hi Coinlogru, oops!!
![image](https://github.com/nboy1/Admob-for-Flash-ANE-GMA21-GLM-/assets/39934716/5784ae62-80c1-430b-971b-93c5be61b50f)


Blocked for security reasons in Gmail

**Comment by nboy1** [comment]
you can check if you used IOS permission config, the GLM ANE is working fine in ios and android just check if something not declared there, try to load Ads in "main" the beginning of loading game then show it in the function you used to reward the user, or make empty project just with loading and showing the ads and follow the documentation to setup things just to debug what's wrong.

**Comment by nboy1** [comment]
i've uploaded the source code you can download it from here 
https://github.com/nboy1/admob-for-flash-android-GMA-21.0.0/blob/main/native_src%20-%20works_27-07-2022_lilili87222.zip
